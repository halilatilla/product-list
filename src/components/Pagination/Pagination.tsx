import { FC } from 'react'
import classnames from 'classnames'
import RcPagination from 'rc-pagination'

import IPagination from './Pagination.types'

const Pagination: FC<IPagination> = ({ className, current, total, pageSize, onChange, hideOnSinglePage }) => {
  return (
    <RcPagination
      current={current}
      total={total}
      pageSize={pageSize}
      hideOnSinglePage={hideOnSinglePage}
      onChange={onChange}
      className={classnames('pagination', className)}
    />
  )
}

export default Pagination
