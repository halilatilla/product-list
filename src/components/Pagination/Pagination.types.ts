export default interface IPagination {
  className?: string
  current?: number
  total?: number
  pageSize?: number
  onChange?: (page: number) => void
  hideOnSinglePage?: boolean
}
