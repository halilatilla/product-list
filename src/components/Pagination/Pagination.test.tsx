import { render } from '@src/test/test-utils'

import Pagination from './Pagination'

describe('Pagination', () => {
  const PROPS = {
    className: '',
    current: 1,
    total: 100,
    pageSize: 10,
    onChange: (page: number) => {},
    hideOnSinglePage: true,
  }
  it.only('renders correctly', () => {
    const { container } = render(<Pagination {...PROPS} />)
    expect(container).toMatchSnapshot()
  })
})
