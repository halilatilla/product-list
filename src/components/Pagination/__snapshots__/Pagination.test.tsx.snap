// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div>
  <ul
    class="rc-pagination pagination pagination"
    unselectable="unselectable"
  >
    <li
      aria-disabled="true"
      class="rc-pagination-prev rc-pagination-disabled"
      title="上一页"
    >
      <button
        aria-label="prev page"
        class="rc-pagination-item-link"
        disabled=""
        type="button"
      />
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-1 rc-pagination-item-active"
      tabindex="0"
      title="1"
    >
      <a
        rel="nofollow"
      >
        1
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-2"
      tabindex="0"
      title="2"
    >
      <a
        rel="nofollow"
      >
        2
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-3"
      tabindex="0"
      title="3"
    >
      <a
        rel="nofollow"
      >
        3
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-4"
      tabindex="0"
      title="4"
    >
      <a
        rel="nofollow"
      >
        4
      </a>
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-5 rc-pagination-item-before-jump-next"
      tabindex="0"
      title="5"
    >
      <a
        rel="nofollow"
      >
        5
      </a>
    </li>
    <li
      class="rc-pagination-jump-next"
      tabindex="0"
      title="向后 5 页"
    >
      <button
        aria-label="next page"
        class="rc-pagination-item-link"
        type="button"
      />
    </li>
    <li
      class="rc-pagination-item rc-pagination-item-10"
      tabindex="0"
      title="10"
    >
      <a
        rel="nofollow"
      >
        10
      </a>
    </li>
    <li
      aria-disabled="false"
      class="rc-pagination-next"
      tabindex="0"
      title="下一页"
    >
      <button
        aria-label="next page"
        class="rc-pagination-item-link"
        type="button"
      />
    </li>
    <li
      class="rc-pagination-options"
    />
  </ul>
</div>
`;
